content/blog/2021-12-07-ml-experiment-versioning.md

```diff
@@ -0,0 +1,238 @@
+---
+title: Don't Just Track Your ML Experiments, Version Them
+date: 2021-12-07
+description: >
+  ML experiment brings together the benefits of traditional code versioning and
+  modern day experiment tracking, super charging your ability to reproduce and
+  iterate on your work.
+descriptionLong: >
+  ML experiment versioning takes experiment tracking to the next level by adding
+  the benefits of version control. Track your parameters and metrics as code,
+  make updates incrementally, and keep everything distributed to share however
+  you want. Iterate on your ML projects at scale in ways that neither
+  traditional software version control nor existing experiment tracking tools
+  can.
+picture:
+pictureComment:
+author: dave_berenbaum
+commentsUrl:
+tags:
+  - MLOps
+  - DVC
+  - Experiments
+  - Experiment Tracking
+  - Experiment Versioning
+---
+
+[Experiment tracking tools](https://dvc.org/doc/use-cases/experiment-tracking)
+help manage machine learning projects where version control tools like Git are
+insufficient. They log parameters and metrics, and they may store binary
+artifacts like input data or model weights so that you can reproduce experiments
+and retrieve results. They enable you to navigate all this meta-information, for
+example with dashboards to organize and compare many experiments.
+
+Git can't manage or compare all that experiment meta-information, but it is
+still better for code. Tools like GitHub make distributed collaboration easy,
+and you can see incremental code changes. Therefore, experiments get split
+between Git for code and experiment tracking tools for meta-information. A link
+may be added in one or the other to keep track.
+
+ML experiment versioning combines experiment tracking and version control:
+
+- **Experiments as code**: Track meta-information in the code repository and
+  version it like code.
+- **Versioned reproducibility**: Save and restore experiment state, and track
+  changes to only execute what's new.
+- **Distributed experiments**: Organize locally and choose what to share,
+  reusing your existing repo setup.
+
+```
```

Comment:
@dmpetrov @jendefig @MKhalusova and anyone else, what do you think about leaving in this ascii art vs. removing it?